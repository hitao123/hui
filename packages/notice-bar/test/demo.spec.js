import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import NoticeBar from '../index.vue';
import NoticeBarDemo from '../demo/index.vue';

describe('NoticeBar component', () => {
  it('should show NoticeBar', () => {
    const warpper = shallowMount(NoticeBar, {
      propsData: {
        text: 'title is long'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show Button Demo', () => {
    expect(renderToString(NoticeBarDemo)).toMatchSnapshot();
  });
});
