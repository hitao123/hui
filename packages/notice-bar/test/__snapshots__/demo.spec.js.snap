// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div>
  <div>
    <div class="h-notice-bar">
      <div class="h-notice-bar__left-icon"><img src="//img.yzcdn.cn/public_files/2017/8/10/6af5b7168eed548100d9041f07b7c616.png"></div>
      <div class="h-notice-bar__wrap">
        <div class="h-notice-bar__content" style="padding-left:0;animation-delay:1s;animation-duration:0s;">足协杯战线连续第2年上演广州德比战，上赛季半决赛上恒大以两回合5-3的总比分淘汰富力。</div>
      </div>
      <!---->
    </div>
  </div>
  <div>
    <div class="h-notice-bar">
      <!---->
      <div class="h-notice-bar__wrap">
        <div class="h-notice-bar__content" style="padding-left:0;animation-delay:1s;animation-duration:0s;">足协杯战线连续第2年上演广州德比战，上赛季半决赛上恒大以两回合5-3的总比分淘汰富力。</div>
      </div>
      <!---->
    </div>
  </div>
  <div>
    <div class="h-notice-bar">
      <div class="h-notice-bar__left-icon"><img src="//img.yzcdn.cn/public_files/2017/8/10/6af5b7168eed548100d9041f07b7c616.png"></div>
      <div class="h-notice-bar__wrap">
        <div class="h-notice-bar__content" style="padding-left:0;animation-delay:1s;animation-duration:0s;">足协杯战线连续第2年上演广州德比战，上赛季半决赛上恒大以两回合5-3的总比分淘汰富力。</div>
      </div> <i class="h-icon h-icon-close h-notice-bar__right-icon"></i>
    </div>
  </div>
  <div>
    <div class="h-notice-bar">
      <!---->
      <div class="h-notice-bar__wrap">
        <div class="h-notice-bar__content" style="padding-left:0;animation-delay:1s;animation-duration:0s;">足协杯战线连续第2年上演广州德比战，上赛季半决赛上恒大以两回合5-3的总比分淘汰富力。</div>
      </div> <i class="h-icon h-icon-arrow h-notice-bar__right-icon"></i>
    </div>
  </div>
</div>
`;
